Development frontend

zzzzwen opened the pull request.

Carl26 committed:
todo text file commit

added a placeholder todo list

Carl26 committed:
real 1st commit

initialize android project

Carl26 committed:
login page commit

framework of login page initialized

Carl26 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into development-frontend

Carl26 committed:
Login in page rework commit

Login in page and registration page has been created.
Implemented basic url connection with server.

Carl26 committed:
Registration & Sign-in commit

separated registration and sign-in process to two different classes.
updated to http connection.
basic implementation of authentication

Carl26 committed:
http builder patch commit

fixed malformed url exception in both http builders

Sheng-Xuan committed:
Update .gitignore file

Sheng-Xuan committed:
ADD movie class.

Sheng-Xuan committed:
ADD travis setting file

Sheng-Xuan committed:
Update Travis CI setting

Sheng-Xuan committed:
ADD third party library 'universal image loader' to support loading images from the Internet.

Sheng-Xuan committed:
ADD parameter 'plot' in movie class

Sheng-Xuan committed:
Update travis.yml file

Sheng-Xuan committed:
ADD images for temporary usage and testing purpose

Carl26 committed:
Registration implementation

registration function implemented
changed registrationHttpBuilder to asynctask

Carl26 committed:
Registration empty details fix

Fixed during registration, email, username, password are not passed
correctly to http builder

Sheng-Xuan committed:
Add library OkHttp and okio for http connection and gson for json file reading.

Sheng-Xuan committed:
Update Movie class to support json convertion.

Sheng-Xuan committed:
Add HomePageActivity as a frame of different fragments. (It is not fully completed)

Sheng-Xuan committed:
Add MovieListFragment, its layout files and adapter. (It is not fully completed)

Sheng-Xuan committed:
Add layout of footer of movie list.

Sheng-Xuan committed:
Add MoviePageFragment and its layouts. (It is not fully completed)

Sheng-Xuan committed:
Add SearchPageFragment and SearchRequestHttpBuilder. (They are not fully completed)

Sheng-Xuan committed:
Update dimens.xml for font setting.

Sheng-Xuan committed:
Update gitignore.

Sheng-Xuan committed:
Stop tracking gradle.xml local setting

Carl26 committed:
Login-in step 1

Implemented creating client on server

Carl26 committed:
Merge branch 'development-frontend' of https://github.com/nus-mtp/movie-pigeon-android into development-frontend

Carl26 committed:
Changed text color in the sign in and registration page

Changed the text color to black

Carl26 committed:
Fixed 401 error crashing login

Implemented a check to see whether the email and password are correct

Sheng-Xuan committed:
Update the async http request in SearchPage.

Sheng-Xuan committed:
Disable the scolling to load more feature, which needs API support, will be implemented in the next version.

Sheng-Xuan committed:
Update movie page layout settings to wrap the content in a new line if it can not be displayed in a single line.

Sheng-Xuan committed:
Add toast in SearchPage, it will be displayed if there is connection error or the search has no results.

Carl26 committed:
Fixed error message in http builders

Carl26 committed:
Login step 2 implementation

Used a cookie manager to maintain session

Carl26 committed:
Merge remote-tracking branch 'origin/development-frontend' into development-frontend

Solving conflicts

Sheng-Xuan committed:
Edit conditions of get movie info based on change in db format.

Sheng-Xuan committed:
Edit SearchRequestHttpBuilder because of merge conflicts.

Carl26 committed:
3rd step of login implementation

Implemented post method to get token

Sheng-Xuan committed:
Remove unused textView Id.

Sheng-Xuan committed:
ADD check for internet connection and empty keyword of search.

Carl26 committed:
Implementaion of token local storage

Right now the token will be stored in a text file for easy testing
purpose

Carl26 committed:
Implemented after login processes

Call the new activity and pass token through the first intent

Carl26 committed:
Receiver class to receive new activity intent

Sheng-Xuan committed:
Update SearchPageHttpBuilder, now the access is taken from signin page.

Carl26 committed:
Refactor registration and sign-in code

Carl26 committed:
Implemented auto login after registration

after registration, a new broadcast is sent to do the sign-in procedures

Carl26 committed:
Implemented go-back function from register page

added a back button in registration page

Carl26 committed:
Added toasts for better user experience and error handling

registration successful toast
wrong email/ password and unable to sign in toast

Carl26 committed:
Implemented auto sign-in if already logged in

auto sign in feature for returning users
token no longer stored in text file though still not encrypted

Sheng-Xuan committed:
UPDATE UI design to improve usability and consistency.

Carl26 committed:
Fixed nullpointerexception in registration and sign-in

Detects whether all required fields are filled in

Carl26 committed:
Fixed error toasts crashing the app

moved toasts inside asynctask doinbackground function to onpostexecute
function

Carl26 committed:
Implemented error handling in registration

catch connection error and user existed error and display them as toasts

Sheng-Xuan committed:
Delete unnecessary lines in AndroidManifest.xml file.

Sheng-Xuan committed:
ADD display of length unit(min) in movie page.

Sheng-Xuan committed:
Fix bug that cannot search movie after auto login.

Sheng-Xuan committed:
Change the default theme setting, remove the title bar by default.

Sheng-Xuan committed:
Add menu icons that will be used in home page.

Sheng-Xuan committed:
Add menu icon that will be used in home page.

Sheng-Xuan committed:
Add display activity to handle intent of displaying movie list, movie detail, etc.

Sheng-Xuan committed:
Add me page fragment and recommendation page fragment to the Home page activity.

Sheng-Xuan committed:
Add displayActivity in Manifest.

Sheng-Xuan committed:
Add homeViewPagerAdapter to help the transition of different fragments in home page.

Sheng-Xuan committed:
Discard function of SearchPageFragment, which is replaced by search view in the home page.

Sheng-Xuan committed:
Update the argument key passed to MovieListFragment.

Sheng-Xuan committed:
Update HomePageActivity to be a container of viewPager.

Sheng-Xuan committed:
Add layout drawables, colors, styles based on the update of layout in HomePage.

Sheng-Xuan committed:
Replace SearchRequestHttpBuilder with ListRequestHttpBuilderSingleton, to make the builder more flexible.

Sheng-Xuan committed:
Add a toolbar in display activity layout.

Sheng-Xuan committed:
Adjust values in UI layouts.

Sheng-Xuan committed:
Update styles and colors.

Sheng-Xuan committed:
Add offset and limit of display in MovieListFragment, and scroll to load more feature.

Sheng-Xuan committed:
Update httpBuilder usage in RecommendationFragment.

Sheng-Xuan committed:
Add display of movie title on the toolbar.

Sheng-Xuan committed:
Add images of public rating source.

Sheng-Xuan committed:
Add images of public rating source.

Sheng-Xuan committed:
Update app icon.

Sheng-Xuan committed:
Add a loading image for posters that are downloading.

Sheng-Xuan committed:
Add public rating object in Movie class.

Sheng-Xuan committed:
Revert "Add public rating object in Movie class."

This reverts commit e2490f61aa0247010698c6fe6b802833b3994ca5.

Sheng-Xuan committed:
Revert "Revert "Add public rating object in Movie class.""

This reverts commit 454295863c0e894c833fc3ad3e14c851d54ab289.

Sheng-Xuan committed:
Add public rating class.

Sheng-Xuan committed:
Redesign movie page, now it displays public ratings and bookmark, rate buttons.

Sheng-Xuan committed:
Change the logic of passing movie info from movie list to movie page, now pass the whole movie object.

Sheng-Xuan committed:
Fix display bug in display activity layout.

Sheng-Xuan committed:
Update movie page layout according to the new design.

Sheng-Xuan committed:
Update movie model to support rating, bookmark and movie count.

Sheng-Xuan committed:
Update RequestBuilder.

Sheng-Xuan committed:
Enable my rating, my bookmark button in ME page.

Sheng-Xuan committed:
Set access token when the app is launched.

Sheng-Xuan committed:
Update logic of movie list and movie page, which are used for search, my ratings and my bookmarks.

Carl26 committed:
Logout functionality commit

Implemented logout function, which deletes the credentials from the sd
card and restart from the sign in page.

Carl26 committed:
Email address check

check correct format of email address in both registration and sign-in
page

Carl26 committed:
Refactored log tag

Extracted out tag in both registration and sign in page

Sheng-Xuan committed:
Add icon for new bookmark display.

Sheng-Xuan committed:
Add Eventbus library

Sheng-Xuan committed:
Update display of bookmark and rating bar.

Sheng-Xuan committed:
Add events that will be used in local movie list updating.

Sheng-Xuan committed:
Add constructor to UserRating and UserBookmark classes.

Sheng-Xuan committed:
Update movie page fragment with new UI and update local list feature.

Sheng-Xuan committed:
Add methods in Movie List Fragment to subscribe the update list events post by movie page.

Sheng-Xuan committed:
Disable landscape mode in all activities.

Sheng-Xuan committed:
Update signin page UI.

Carl26 committed:
Change username and password commit

Implemented change username and password function

Sheng-Xuan committed:
Add testcases for RequsetHttpBuilder and MovieListAdapter.

Sheng-Xuan committed:
Delete travis CI

Sheng-Xuan committed:
Add circle CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to Fix CI

Sheng-Xuan committed:
Try to Fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix circle CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Try to fix CI

Sheng-Xuan committed:
Fix a bug that the list view title is displayed incorrectly

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/res/layout/activity_main.xml:
```diff
@@ -11,9 +12,85 @@
+        android:textColor="#77BEDC"
+        android:gravity="center"
+        android:layout_marginTop="20dp"
+        android:id="@+id/textView"
```
Remember to set a proper id of each view.

Carl26 replied:
This is outdated and will not occur in the current build.

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/Receiver.java:
```diff
@@ -38,6 +38,9 @@ public void onReceive(Context context, Intent intent) {
             signInDetails[1] = passwordSignin[0];
             SignInHttpBuilder sBuilder = new SignInHttpBuilder(context);
             sBuilder.execute(signInDetails);
+        } else if (intent.getAction().equals("UserUpdate")) {
```
This is case, using switch/case may improve readability of code.

Carl26 replied:
fixed

Carl26 commented on frontend/Movie_Pigeon/app/build.gradle:
```diff
@@ -0,0 +1,49 @@
+    }
+}
+
+dependencies {
```
Some dependencies may be optimized by manual coding as high amount of dependencies may require large amount of space when installing the app.

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/MeFragment.java:
```diff
@@ -65,6 +66,18 @@ public void onClick(View v) {
+
+            @Override
+            public void onClick(View v) {
+                Log.e(TAG, "Setting Button clicked");
```
This should be Log.i instead of Log.e, this is not a error log.

Carl26 replied:
fixed

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/RegistrationHttpBuilder.java:
```diff
@@ -52,17 +53,17 @@ private void request(String query) {
 
             int status = connection.getResponseCode();
-            Log.e("rHttpBuilder", "response status transactionId is " + status);
+            Log.e(TAG, "response status transactionId is " + status);
```
Same problem, Log.e should not be used.

Carl26 replied:
fixed

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/RegisterPage.java:
```diff
@@ -57,7 +57,8 @@ public void onClick(View v) {
                         Toast toast = Toast.makeText(mContext, "Passwords entered don't match!", Toast.LENGTH_SHORT);
                         toast.show();
-
+                    } else if (!email[0].contains("@")) {
```
May implement a better checker with regex, now "@@" which is not a valid email will be pass this check.

Carl26 replied:
fixed

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/MeFragment.java:
```diff
@@ -53,9 +57,36 @@ public void onClick(View v) {
+            getActivity().startActivity(loadStartActivity);
+            Log.e(TAG, "Finishing current activity");
+            getActivity().finish();
+        } else {
```
'else' here may never run? Failed to logout may happen when the app does not have the permission to delete/read file. 'else' here should handle IOException instead?

Carl26 replied:
Invalid: There is no IOException here in else since it only detects credential file not available but not trying to delete it. It is more like the user accidentally deleted the file while our app is running so when he tries to logout, the file is already gone and will not be deleted twice. 

New: yes IOException indeed occurs if file is still there but the detection should take place in the previous method.

I will make a change to handle the problem of still existing credential file.

Update 2: so the file.delete() function will return a boolean. if false, it means it fails to delete and will not throw an exception to catch. This else clause will still run nonetheless.

Carl26 replied:
updated toast to reflect the error better

Carl26 commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/MovieListFragment.java:
```diff
@@ -0,0 +1,227 @@
+                }
+                return null;
+            } catch (IOException e) {
+                status = NO_INTERNET;
```
This IOException may also occur when our server is down thus it does not necessarily mean the problem lies within the user's control.

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/RegistrationHttpBuilder.java:
```diff
@@ -53,19 +57,30 @@ private void request(String query) {
         } catch (IOException e) {
             e.printStackTrace();
+            connectionError = true;
             System.out.print("Unable to connect to server");
```
Use Log instead of sysout

Carl26 replied:
fixed

Carl26 commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/MoviePageFragment.java:
```diff
@@ -0,0 +1,319 @@
+        }
+    }
+
+    private void setRatingsOthersSetGone(String rating, TextView score, ImageView icon) {
```
Not very consistent in naming when compared to the one right above. i.e. Otherwise vs Others

Sheng-Xuan replied:
Fixed

Carl26 commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/MoviePageFragment.java:
```diff
@@ -0,0 +1,319 @@
+        @Override
+        protected Integer doInBackground(String... params) {
+            Request request = null;
+            switch (params[0]) {
```
Maybe save params[X] as new variables inside MyTask class to show what they do for easier reading and debugging purposes?

Sheng-Xuan replied:
Fixed

Carl26 commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/eventCenter/AddMovieToMovieListEvent.java:
```diff
@@ -0,0 +1,17 @@
+ * Created by SHENGX on 2017/2/21.
+ */
+
+public class AddMovieToMovieListEvent {
```
The series of add, delete and update movie classes may be integrated into one class or as an interface to be implemented in the usage class.

Carl26 reviewed (approved):
Approved with some minor tweaks suggestions.

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/RegisterPage.java:
```diff
@@ -15,21 +16,19 @@
 class RegisterPage {
 
     RegisterPage(Context mContext, Activity mActivity, View signin) {
+        final String[] email = new String[1];
```
Should just use String instead of String array of size one? They are the same right?

Carl26 replied:
fixed

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/Receiver.java:
```diff
@@ -0,0 +1,27 @@
+ * Created by Guo Mingxuan on 11/2/2017.
+ */
+
+class Receiver extends BroadcastReceiver {
```
Should change specify the name of this class.

Carl26 replied:
Meaning change Receiver to MyReceiver or something?

Carl26 replied:
fixed

Carl26 committed:
Registration page optimization

Used string instead of string array to store data.
Removed requiring view in the constructor as it is never used.

Sheng-Xuan committed:
Refractor code in Movie Page fragment.

Sheng-Xuan committed:
Refactor code in Recommendation fragment.

Carl26 committed:
Updated email checking in registration

Now email checks:
1. only 1 @ sign and @ is a must
2. at least length of 3 as a@b

Carl26 committed:
Refactored Receiver class naming to GlobalReceiver

Carl26 committed:
Updated GlobalReceiver class

Use switch cases for better runtime.
Use String instead of String[] to store email and password.

Carl26 committed:
Fixed incorrect usage of Log.e to Log.i

Carl26 committed:
Updated toast of unable to logout error

Carl26 committed:
Updated failed to logout error#2

Carl26 committed:
Fixed usage of system.out.print by changing it to log.i
